packages/core/src/routes/phrase.ts

```diff
@@ -0,0 +1,107 @@
+import { LanguageKey, languageKeyGuard } from '@logto/core-kit';
+import resource from '@logto/phrases-ui';
+import { CustomPhrases } from '@logto/schemas';
+import { adminConsoleApplicationId, adminConsoleSignInExperience } from '@logto/schemas/lib/seeds';
+import deepmerge from 'deepmerge';
+import { Provider } from 'oidc-provider';
+
+import detectLanguage from '@/i18n/detect-language';
+import koaGuard from '@/middleware/koa-guard';
+import { findAllCustomLanguageKeys, findCustomPhraseByLanguageKey } from '@/queries/custom-phrase';
+import { findDefaultSignInExperience } from '@/queries/sign-in-experience';
+
+import { AnonymousRouter } from './types';
+
+const getLanguageInfo = async (applicationId: unknown) => {
+  if (applicationId === adminConsoleApplicationId) {
+    return adminConsoleSignInExperience.languageInfo;
+  }
+
+  const { languageInfo } = await findDefaultSignInExperience();
+
+  return languageInfo;
+};
+
+const isBuiltInLanguageKey = (key: string): key is LanguageKey =>
+  Object.keys(resource).includes(key) && languageKeyGuard.safeParse(key).success;
```

Comment:
`languageKeyGuard.safeParse(key).success` this code seems redundant and maybe cause errors in the future.
Because our built-in language comes from `resource`, and the `languageKey` represent the supported language keys (including custom languages).